packages/devextreme/js/__internal/ui/html_editor/ui/aiDialog.ts

```diff
@@ -63,37 +76,43 @@ export interface AIDialogResult {
 }
 
 export default class AIDialog extends BaseDialog<AIDialogResult> {
-  private _dialogState: DialogState = DialogState.Initial;
-
-  private _isAskAICommandSelected = false;
+  private _abort?: () => void;
 
   private _aiIntegration: AIIntegration;
 
-  private _commandsMap: CommandsMap = {};
-
-  private _currentCommand?: string;
+  private _askAIPrompt = '';
 
-  private _currentOption?: string;
+  private _commandChangeSuppressed = false;
 
   private _commandOptionsList?: string[];
 
-  private _selectedText = '';
+  private _commandSelectBox!: dxSelectBox;
 
-  private _resultText = '';
+  private _commandsMap: CommandsMap = {};
 
-  private _askAIPrompt = '';
+  private _currentCommand?: AICommandNameExtended;
+
+  private _currentOption?: string;
+
+  private _dialogState: DialogState = DialogState.Initial;
 
   private _getCustomCommandPrompt?: AICustomCommand['prompt'];
 
-  private _commandSelectBox!: dxSelectBox;
+  private _isAICommandExecuted = false;
 
-  private _optionSelectBox!: SelectBox;
+  private _isAskAICommandSelected = false;
 
-  private _resultTextArea!: TextArea;
+  private _loadIndicator?: LoadIndicator;
+
+  private _optionSelectBox!: SelectBox;
 
   private _promptTextArea!: TextArea;
 
-  private _commandChangeSuppressed = false;
+  private _resultText = '';
```

Comment:
Do we need this while we also have `_resultTextArea`? I presume we can `this._resultTextArea.option('value')` where need it and make -1 state variable

Reply:
Ruslan's idea was to update all states in one place, so this variable appeared. But we have moved away from this concept.

Reply:
[68ca6bbeba3b204e933d3eb10ee51cc141b8547e]